### ISSUE TITLE
Allow setting configuration using UEFI variables

### ISSUE THREAD

**Issue by sftim** [EmilyShepherd/kios-uefi#1]
_As a cluster operator
I want to set a UEFI variable to control the DNS resolver configuration in Kios
So that I can use custom DNS settings
And I do not need to bake a new image for each different set of DNS servers._

That, basically. Some things need to be right very early in boot, and DNS is often one of those things.

**Comment by EmilyShepherd** [comment]
Agreed - I think it would be sensible to have some better form of UEFI variable support. I _suspect_ I'll add this support via an optional bootstrap pod (a la kios-aws' bootstrap container) but open to other suggestions

**Comment by EmilyShepherd** [comment]
The alpha version of the uefi bootstrap container is almost ready now. This project can configure the following, via UEFI variables:

- Cluster CA
- Instance Hostname
- Api-Server Endpoint
- Node Labels
- Nameservers

As part of the work to do this, I split most of the generic stuff out of the [kios-aws](https://github.com/EmilyShepherd/kios-aws) project into a more generalised [go-sdk](https://github.com/EmilyShepherd/kios-go-sdk), which now both this project and the aws project use. Hopefully will make life easy if and when other bootstrap mechanisms are dreamed up :smile:
